### DISCUSSION TITLE
Landscape Errors

### DISCUSSION THREAD

**Discussion by ee20gr** [RangeShifter/RangeshiftR-tutorials#43] [Q&A]
I am trying to run RangeShiftR for the first time. I have created a landscape habitat map with 2 habitat types (codes 1 & 2) and a species distribution map with cell values of 4 for where the species is present and 0 where they are not (nodata values). However, when I try to run the simulation I get the following error messages:

Run Simulation(s) with random seed ...
LandscapeFile headers OK: C:/MAP1_RUN/Inputs/Landscape_Control_Copy.asc
Species Distribution map headers OK: C:/MAP1_RUN/Inputs/PM_SpDist.asc
Found invalid habitat code.

Error reading landscape 0 - aborting
At (x,y) = 1,2119 :
*** Corrupted file stream in C:/MAP1_RUN/Inputs/PM_SpDist.asc
Too few entries? Unsuppoerted file encoding? (You might try to use a different one, like UTF-8.)

Error reading initial distribution for landscape 0 - aborting

I am not sure what to do to fix these errors, can anyone help? Thanks

**Comment by JetteReeg** [comment]
Hi, 

in general it would be great if you could also provide a minimal working example, which reproduces your problem (see guidelines of how to ask for help, e.g. guidelines given by [StackOverflow](https://stackoverflow.com/help/how-to-ask). 

From what I get out of the information you provided, I guess your error occurs as you have values of 4 and 0 (is that also your nodata value?) for the initial distribution map. 

However, as you can find in the `?ImportedLandscape` documentation:

> Each cell of the species distribution map must contain either 0 (species absent or not recorded) or 1 (species present).

So all cells which have the habitat code 1 and 2 in you landscape file should have either 0 (species absent) and 1 (species present). Cells with no data should have a value of e.g. -999 in both the landscape and the initial distribution file. 

You can also check the files of the [Tutorial 1](https://rangeshifter.github.io/RangeshiftR-tutorials/tutorial_1.html) which uses an initial distribution file.

If you use RangeShiftR for the first time, I also recommend youfirst study the provided documentation ([RangeShifter manual](https://raw.githubusercontent.com/RangeShifter/RangeShifter-software-and-documentation/master/RangeShifter_v2.0_UserManual.pdf), [RangeShiftR tutorial](https://rangeshifter.github.io/RangeshiftR-tutorials), R help pages) before setting up your first own RangeShiftR model (if you haven't done so already).
